Edits for -01 and WGLC preparation. (#3)
tlswg/draft-ietf-tls-ticketrequest: draft-ietf-tls-ticketrequests.md

--- abstract

TLS session tickets enable stateless connection resumption for clients without
server-side, per-client state. Servers vend an arbitrary number of session tickets
to clients, at their discretion, upon connection establishment. Clients store and
use tickets when resuming future connections. This document describes a mechanism by
which clients may specify the desired number of tickets needed for future connections.

> This extension aims to remove server ambiguity surrounding ticket count while simultaneously

Reviewer: This early in the document, it's a bit unclear what "server ambiguity surrounding ticket count" means.

Author: Hmm, I'm not sure adding more text is appropriate here. Do you have an alternative way to phrase this?

Reviewer: Yeah, I agree we don't want more text, but a clearer statement would be nice. How about:
`This extension aims to provide a means for servers to determine the number of tickets to generate, while simultaneously...`

Author: I also like it :) I went with another variant: "This extension aims to provide a means for servers to determine the number of tickets to generate as a means of reducing ticket waste, while simultaneously priming clients for future connection attempts."

reducing ticket waste and priming clients for future connection attempts.

--- middle

# Introduction

As per {{RFC5077}}, and as described in {{RFC8446}}, TLS servers send clients an arbitrary
number of session tickets at their own discretion in NewSessionTicket messages. There are

> two limitations with this design. First, servers typically guess (or hard-code) the number

Reviewer: In what way do the servers guess? Do you mean that they guess the number that the client will need to use?

Author: They guess "the number of tickets" vended.

Reviewer: Guess implies that there is a correct answer. This text doesn't clearly define whose correct answer that is. The server just does what it wants for itself, it doesn't really guess. If it is guessing at all, it's guessing what the client would want, right?

Reviewer: How about `servers typically determine some (often hard-coded) number of tickets...`

Author: I like it! I went with a variant: "servers choose some (often hard-coded) number.."

of tickets vended per connection. Second, clients do not have a way of expressing their
desired number of tickets, which may impact future connection establishment.
For example, clients may open multiple TLS connections to the same server for HTTP,
or may race TLS connections across different network interfaces. The latter is especially

> useful in transport systems that implement Happy Eyeballs V2 {{RFC8305}}. Since clients control

Reviewer 2: I would just say Happy Eyeballs {{RFC8305}} because the property here isn't specific to v2.

Author: Fixed!

connection concurrency and resumption, a standard mechanism for requesting more than one
ticket is desirable.

This document specifies a new TLS extension -- "ticket_request" -- that may be used
by clients to express their desired number of session tickets. Servers may use this
extension as a hint of the number of NewSessionTicket messages to vend.
This extension is only applicable to TLS 1.3 {{!RFC8446}}, DTLS 1.3 {{!I-D.ietf-tls-dtls13}},

> and future versions of thereof.

Reviewer 2: Is `of thereof` correct english? I would have written `of both`

Author: Nope, typo! Fixed.

--- abstract

TLS session tickets enable stateless connection resumption for clients without
server-side, per-client state. Servers vend an arbitrary number of session tickets
to clients, at their discretion, upon connection establishment. Clients store and
use tickets when resuming future connections. This document describes a mechanism by
which clients may specify the desired number of tickets needed for future connections.
This extension aims to provide a means for servers to determine the number of tickets

> to generate as a means of reducing ticket waste, while simultaneously priming clients

Reviewer: Nit: I don't love having `a means...` twice in the sentence. I would change this second instance to `...to generate in order to reduce ticket waste,`

Author: Fixed!

> to generate as a means of reducing ticket waste, while simultaneously priming clients

Author: ```suggestion
to generate in order to reduce ticket waste, while simultaneously priming clients
```

# IANA Considerations

IANA is requested to Create an entry, ticket_request(TBD), in the existing registry
for ExtensionType (defined in {{RFC8446}}), with "TLS 1.3" column values being set to
"CH", and "Recommended" column being set to "Yes".

# Security Considerations

Ticket re-use is a security and privacy concern. Moreover, clients must take care when pooling
tickets as a means of avoiding or amortizing handshake costs. If servers do not rotate session
ticket encryption keys frequently, clients may be encouraged to obtain
and use tickets beyond common lifetime windows of, e.g., 24 hours. Despite ticket lifetime
hints provided by servers, clients SHOULD dispose of pooled tickets after some reasonable
amount of time that mimics the ticket rotation period.

# Acknowledgments

The authors would like to thank David Benjamin, Eric Rescorla, Nick Sullivan, Martin Thomson,

> and other members of the TLS WG for discussions on earlier versions of this draft.

Reviewer 2: nit: spell out working group

Author: Fixed.
